docs/conf.py

```diff
@@ -57,3 +64,9 @@
 ]
 
 suppress_warnings = ["myst.xref_missing"]
+
+
+def setup(app: Sphinx) -> None:
```

Comment:
Don't add `setup()` here but put it into an extension under `_ext/`.

Also, it should return a dict.